client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -1279,12 +1318,21 @@ impl<Block: BlockT> GossipValidator<Block> {
 		let mut broadcast_topics = Vec::new();
 		let mut peer_reply = None;
 
+		// Message name for Prometheus metric recording.
+		let message_name;
+
 		let action = {
```

Comment:
I think this would be better like 
```suggestion
		let (action, message_name) = {
```

Reply:
Are you suggesting the following?

```diff
diff --git a/client/finality-grandpa/src/communication/gossip.rs b/client/finality-grandpa/src/communication/gossip.rs
index 89c507f39..8751f4c9b 100644
--- a/client/finality-grandpa/src/communication/gossip.rs
+++ b/client/finality-grandpa/src/communication/gossip.rs
@@ -1314,21 +1314,13 @@ impl<Block: BlockT> GossipValidator<Block> {
                let mut broadcast_topics = Vec::new();
                let mut peer_reply = None;
 
-               // Message name for Prometheus metric recording.
-               let message_name;
-
-               let action = {
+               let (action, message_name) = {
                        match GossipMessage::<Block>::decode(&mut data) {
-                               Ok(GossipMessage::Vote(ref message)) => {
-                                       message_name = Some("vote");
-                                       self.inner.write().validate_round_message(who, message)
-                               },
-                               Ok(GossipMessage::Commit(ref message)) => {
-                                       message_name = Some("commit");
-                                       self.inner.write().validate_commit_message(who, message)
-                               },
+                               Ok(GossipMessage::Vote(ref message)) =>
+                                       (self.inner.write().validate_round_message(who, message), Some("vote")),
+                               Ok(GossipMessage::Commit(ref message)) =>
+                                       (self.inner.write().validate_commit_message(who, message), Some("commit")),
                                Ok(GossipMessage::Neighbor(update)) => {
-                                       message_name = Some("neighbor");
                                        let (topics, action, catch_up, report) = self.inner.write().import_neighbor_message(
                                                who,
                                                update.into_neighbor_packet(),
@@ -1340,14 +1332,12 @@ impl<Block: BlockT> GossipValidator<Block> {
 
                                        broadcast_topics = topics;
                                        peer_reply = catch_up;
-                                       action
+                                       (action, Some("neighbor"))
                                }
                                Ok(GossipMessage::CatchUp(ref message)) => {
-                                       message_name = Some("catch_up");
-                                       self.inner.write().validate_catch_up_message(who, message)
+                                       (self.inner.write().validate_catch_up_message(who, message), Some("catch_up"))
                                },
                                Ok(GossipMessage::CatchUpRequest(request)) => {
-                                       message_name = Some("catch_up_request");
                                        let (reply, action) = self.inner.write().handle_catch_up_request(
                                                who,
                                                request,
@@ -1355,15 +1345,14 @@ impl<Block: BlockT> GossipValidator<Block> {
                                        );
 
                                        peer_reply = reply;
-                                       action
+                                       (action, Some("catch_up_request"))
                                }
                                Err(e) => {
-                                       message_name = None;
                                        debug!(target: "afg", "Error decoding message: {}", e.what());
                                        telemetry!(CONSENSUS_DEBUG; "afg.err_decoding_msg"; "" => "");
 
                                        let len = std::cmp::min(i32::max_value() as usize, data.len()) as i32;
-                                       Action::Discard(Misbehavior::UndecodablePacket(len).cost())
+                                       (Action::Discard(Misbehavior::UndecodablePacket(len).cost()), None)
                                }
                        }
                };
```

I would not prefer this code style.

1. I see `action` as actual business logic whereas `message_name` a meta logic, combining the two in one tuple removes the differentiation. 

2. Just by looking at a line like `(action, Some("catch_up_request"))` the reader can not imply what the second element of the tuple is.

Do you feel strongly about this @expenses?

Reply:
No, I don't.